presto-main/src/main/java/com/facebook/presto/sql/planner/plan/MergeRemoteSourceNode.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.spi.block.SortOrder;
+import com.facebook.presto.sql.planner.Symbol;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@Immutable
+public class MergeRemoteSourceNode
+        extends RemoteSourceNode
+{
+    private final List<Symbol> orderBy;
+    private final Map<Symbol, SortOrder> orderings;
+
+    @JsonCreator
+    public MergeRemoteSourceNode(
+            @JsonProperty("id") PlanNodeId id,
+            @JsonProperty("sourceFragmentIds") List<PlanFragmentId> sourceFragmentIds,
+            @JsonProperty("outputs") List<Symbol> outputs,
+            @JsonProperty("orderBy") List<Symbol> orderBy,
+            @JsonProperty("orderings") Map<Symbol, SortOrder> orderings)
+    {
+        super(id, sourceFragmentIds, outputs);
+        this.orderBy = orderBy;
+        this.orderings = orderings;
+    }
+
+    public MergeRemoteSourceNode(
+            PlanNodeId id,
+            PlanFragmentId sourceFragmentId,
+            List<Symbol> outputs,
+            List<Symbol> orderBy,
+            Map<Symbol, SortOrder> orderings)
+    {
+        this(id, ImmutableList.of(sourceFragmentId), outputs, orderBy, orderings);
+    }
+
+    @JsonProperty("orderBy")
+    public List<Symbol> getOrderBy()
+    {
+        return orderBy;
+    }
+
+    @JsonProperty("orderings")
+    public Map<Symbol, SortOrder> getOrderings()
+    {
+        return orderings;
+    }
+
+    @Override
+    public <C, R> R accept(PlanVisitor<C, R> visitor, C context)
+    {
+        return visitor.visitMergeRemoteSource(this, context);
+    }
+
+    @Override
+    public PlanNode replaceChildren(List<PlanNode> newChildren)
+    {
+        checkArgument(newChildren.isEmpty(), "newChildren is not empty");
+        return this;
+    }
+}
```

Comment:
You might need to override equals and hashCode (since RemoteSourceNode might define it)